wp-content/civi-extensions/goonjcustom/goonjcustom.php

```diff
@@ -125,6 +125,32 @@ function goonjcustom_civicrm_buildForm( $formName, $form ) {
 					$form->removeElement( $field );
 				}
 			}
+
+            CRM_Core_Region::instance('page-body')->add(array(
+                'script' => "
+                    CRM.$(function($) {
+                        function updateCustomGroupVisibility() {
+                            var selectedText = $('#status_id').find('option:selected').text();
+                            var customGroup = $('.custom-group-INDUCTION_DETAILS');
+
+                            if (selectedText === 'Completed') {
+                                customGroup.show();
+								// TODO - need to add logic to handle required fields
+                            } else {
+                                customGroup.hide();
+								// TODO - need to add logic to handle required fields
+                            }
+                        }
+                        $('#status_id').change(function() {
+						var selectedValue = $(this).val();
+                            updateCustomGroupVisibility();
+                        });
+
+                        // Initial check to show/hide the custom group and update required constraints
+                        setTimeout(updateCustomGroupVisibility, 500);
```

Comment:
Is this a hack or something?

Reply:
Yes, it takes some time for form to load, this logic need some delay before it get executed with the available elements.

Reply:
Can you add this as a comment?

Reply:
Done